lib/teleterm/daemon/daemon.go

```diff
@@ -202,22 +206,28 @@ func (s *Service) RemoveGateway(ctx context.Context, gatewayURI string) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	s.removeGateway(gateway)
+	if err := s.removeGateway(gateway); err != nil {
+		return trace.Wrap(err)
+	}
 
 	return nil
 }
 
 // removeGateway assumes that mu is already held by a public method.
-func (s *Service) removeGateway(gateway *gateway.Gateway) {
-	gateway.Close()
+func (s *Service) removeGateway(gateway *gateway.Gateway) error {
+	if err := gateway.Close(); err != nil {
+		return trace.Wrap(err)
+	}
 
 	// remove closed gateway from list
 	for index := range s.gateways {
 		if s.gateways[index] == gateway {
 			s.gateways = append(s.gateways[:index], s.gateways[index+1:]...)
```

Comment:
Why isn't `gateways` a hashset of `*Gateway`?

Reply:
I don't think there's any particular reason for why it has to be just an array rather than a hashset, it was written as so and we just didn't change it. I'll make a separate PR to address that, you're not the first person who asked this question. ;)